app/api/recommend.py

```
@@ -110,30 +110,51 @@ def startup_event():
         else: 
-            logger.info(f"[CACHE][MISS] key={cache_key_masked}")
+            logger.info(f"[CACHE][MISS] key=%s", cache_key_masked, extra={
+                "traceId": trace_id,
+                "userId": user_id,
+                "endpoint": endpoint,
+                "status": 200,
+                "latencyMs": int((time.perf_counter() - start_time)*1000)
+            })
     except Exception as e:
-        logger.error(f"[CACHE][ERROR] stage=get key={cache_key_masked} err={e}", exc_info=True)
+        logger.error(f"[CACHE][ERROR] key=%s err=%s", cache_key_masked, str(e), extra={
```

Ruff (0.12.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)